include/hid/robot_control.h

```diff
@@ -0,0 +1,222 @@
+#pragma once
+
+// BoB robotics includes
+#include "common/circstat.h"
+#include "hid/joystick.h"
+#include "robots/type_traits.h"
+
+// Standard C++ includes
+#include <algorithm>
+#include <array>
+#include <type_traits>
+
+// Standard C includes
+#include <cmath>
+
+#define DEFAULT_DEAD_ZONE 0.25f
+
+namespace BoBRobotics {
+namespace HID {
+template<class RobotType>
+void
+drive(RobotType &robot, const JoystickBase<JAxis, JButton> &joystick)
+{
+    drive(robot, joystick, DEFAULT_DEAD_ZONE);
+}
+
+template<class AckermannType,
```

Comment:
I wonder if this could be restructured a little to reduce the duplication of ``enable_if`` horror:
```c++
template <typename AckermannType,
    std::enable_if_t<Robots::isAckermann<AckermannType>::value, int> = 0>
struct RobotControl
{
     static constexpr auto useAxes();

     static void addJoystick(...);

     static void drive(...);
};
```

Reply:
Good idea!

Reply:
Although I've just realised that the compiler won't be able to infer the type from the arguments passed to these member functions, so you will end up with horrors like this:
```c++
        HID::RobotControl<ROBOT_TYPE>::drive(robot, joystick, 0.25f);
```

You *could* get round this somewhat by having a ``using RobotControl = HID::RobotControl<ROBOT_TYPE>;`` at the start of your method but it still seems a bit gross.

Reply:
Oh yes - shit! That is always the problem with templated classes/structs - adding wrappers would undo any of the tidiness improvement many times over so probably best to leave as is

Reply:
Cool. So are we good to merge then? I think I've addressed everything...

Reply:
Totally fine to merge!